multimodal/src/autogluon/multimodal/utils/matcher.py

```diff
@@ -0,0 +1,280 @@
+    response_fusion_model_name = [n for n in response_config.model.names if n.lower().startswith(FUSION)]
+    assert len(response_fusion_model_name) <= 1
+    if len(response_fusion_model_name) == 1:
+        response_fusion_model_name = response_fusion_model_name[0]
+
+    logger.debug(f"single model names: {list(single_models.keys())}")
+    logger.debug(f"query fusion model name: {query_fusion_model_name}")
+    logger.debug(f"response fusion model name: {response_fusion_model_name}")
+
+    # use shallow copy to create query single models
+    query_single_models = []
+    for model_name in query_model_names:
+        model = copy.copy(single_models[model_name[:-6]])  # cut off _query
+        model.prefix = model_name
+        query_single_models.append(model)
+
+    # use shallow copy to create response single models
+    response_single_models = []
+    for model_name in response_model_names:
+        model = copy.copy(single_models[model_name[:-9]])  # cut off _response
+        model.prefix = model_name
+        response_single_models.append(model)
+
+    if len(query_single_models) == 1:
+        query_model = query_single_models[0]
+    else:
+        if initialized:
+            query_model = query_fusion_model
+            query_model.model = nn.ModuleList(query_single_models)
+        else:
+            query_model = query_fusion_model(models=query_single_models)
+
+        query_model.prefix = query_fusion_model_name
+
+    if len(response_single_models) == 1:
+        response_model = response_single_models[0]
+    else:
+        if share_fusion:
+            response_model = copy.copy(query_model)  # copy query_model rather than query_fusion_model
+            response_model.model = nn.ModuleList(response_single_models)
+        else:
+            if initialized:
+                response_model = response_fusion_model
+                response_model.model = nn.ModuleList(response_single_models)
+            else:
+                response_model = response_fusion_model(models=response_single_models)
+
+        response_model.prefix = response_fusion_model_name
+
+    return query_model, response_model
+
+
+def is_share_fusion(
+    query_model_names: List[str],
+    response_model_names: List[str],
+):
+    """
+    Check whether the query and response models share the same fusion part.
+
+    Parameters
+    ----------
+    query_model_names
+        Names of single models in the query late-fusion model.
+    response_model_names
+        Names of single models in the response late-fusion model.
+
+    Returns
+    -------
+    Whether to share the same fusion part.
+    """
+    query_model_names = [n for n in query_model_names if not n.lower().startswith(FUSION)]
+    response_model_names = [n for n in response_model_names if not n.lower().startswith(FUSION)]
+    return sorted(query_model_names) == sorted(response_model_names)
+
+
+def create_siamese_model(
```

Comment:
it can be kind of difficult to distinguish at the first glance create_siamese_model and build_siamese_network

Reply:
Changed `build_siamese_network ` to `make_siamese` be distinguishable.